core/imageroot/usr/local/agent/pypkg/agent/__init__.py

```diff
@@ -474,3 +479,60 @@ def get_smarthost_settings(rdb):
         }
     return data
 
+def http_route_in_use(domain='127.0.0.1', path='', expectation=404):
+    """ we search if the domain (foo.com) or the webpath (/foo) is used or not by a service
+        return true if the route is used, false is the route is not used"""
+
+    # We cannot use the domain of the server else 
+    # we cannot reach the cluster-admin
+    if domain == socket.getfqdn():
+        return True
+
+    http = 0
+    https = 0
+
+    # do not verify the certificate
+    ctx = ssl.create_default_context()
+    ctx.check_hostname = False
+    ctx.verify_mode = ssl.CERT_NONE
+    
+    # try on http
+    try:
+        req = urllib.request.Request('http://127.0.0.1'+path)
+        req.add_header('Host', domain)
+        urllib.request.urlopen(req, context=ctx)
+    except urllib.error.HTTPError as e:
+        http = e.code
+
+    # try on https
+    try:
+        req = urllib.request.Request('https://127.0.0.1'+path)
+        req.add_header('Host', domain)
+        urllib.request.urlopen(req, context=ctx)
+    except urllib.error.HTTPError as d:
+        https = d.code
+
+    if http == expectation and  https == expectation:
+        # there is no website on http or https -> OK
+        return False
+    else:
+        # path exists -> nok
+        return True
```

Comment:
That's all...

```suggestion
def http_route_in_use(domain=None, path='', expectation=404):
    """ we search if the domain (foo.com) or the webpath (/foo) is used or not by a service
        return true if the route is used, false is the route is not used"""
    try:
        req = urllib.request.Request('http://127.0.0.1'+path)
        if domain:
            req.add_header('Host', domain)
        urllib.request.urlopen(req, context=ctx)
    except urllib.error.HTTPError as e:
        if e.code == expectation:
            # there is no website on http or https -> OK
            return False

    return True
```

Reply:
...I guess the urlopen() function could also succeed without raising any exception. What happens with status code 200?

Reply:
yes only 404 is the return that the route does not exist

Reply:
Edit: added `method="HEAD"` to avoid fetching page contents